src/nodeClient.js

```diff
@@ -1,93 +1,64 @@
-const Web3 = require('web3');
+const ethers = require('ethers');
 const { helpers } = require('leap-core');
 
 const { formatHostname, advanceBlocks, sleep } = require('./helpers');
 
 let idCounter = 0;
 
-class Node {
+class Node extends helpers.LeapEthers {
   constructor(hostname, jsonrpcPort) {
-    this.id = idCounter++;
-    this.web3 = helpers.extendWeb3(new Web3(formatHostname(hostname, jsonrpcPort)));
-  }
+    const provider = new ethers.providers.JsonRpcProvider(formatHostname(hostname, jsonrpcPort))
+    super(provider);
 
-  async send(method, params) {
-    return await new Promise(
-      (resolve, reject) => {
-        this.web3.currentProvider.send(
-          { jsonrpc: '2.0', id: 42, method: method, 'params': params },
-          (err, res) => { if (err) { return reject(err); } resolve(res); }
-        );
-      }
-    );
+    this.id = idCounter++;
   }
 
   async sendTx(tx) {
-    // workaround different transaction hashes (leap-core bug)
-    let txHash;
-
-    try {
-      // web3 hangs here on invalid txs, trying to get receipt?
-      // await this.web3.eth.sendSignedTransaction(tx.hex());
-      const resp = await new Promise(
-        (resolve, reject) => {
-          this.web3.currentProvider.send(
-            { jsonrpc: '2.0', id: 42, method: 'eth_sendRawTransaction', 'params': [tx.hex()] },
-            (err, res) => { if (err) { return reject(err); } resolve(res); }
-          );
-        }
-      );
-      txHash = resp.result;
-    } catch(e) {
-      console.log(e);
-    }
-
-    if (txHash !== tx.hash()) {
-      console.warn(`txHash(${txHash}) from rpc does not match local tx.hash(${tx.hash()})`);
-    }
+    return helpers.sendSignedTransaction(this.provider, tx.hex());
+  };
 
-    let rounds = 50;
-    while (rounds--) {
-      let res = await this.web3.eth.getTransaction(txHash)
+  async getBalance(addr) {
+    return this.provider.getBalance(addr).then(res => Number(res));
+  }
 
-      if (res && res.blockHash) {
-        return;
-      }
+  async getBlock(val, includeTxs) {
+    let method = 'eth_getBlockByNumber';
 
-      // wait ~100ms
-      await new Promise((resolve) => setTimeout(() => resolve(), 100));
+    if (typeof val === 'string' && val.startsWith('0x')) {
+      method = 'eth_getBlockByHash';
     }
 
-    throw new Error('transaction not included in time');
-  };
+    return this.provider.send(method, [val, includeTxs]);
+  }
 
-  async getBalance(addr) {
-    return this.web3.eth.getBalance(addr).then(res => Number(res));
+  async getTransaction(...args) {
+    // Do it manually here because we sometimes get 'encoding' errors (missing fields returned from rpc?)
```

Comment:
👍 

yes, ethers is checking responses MUCH stricter than web3 (if web3 checks anything at all 😄) — we have to return [a lot of dummies](https://github.com/leapdao/leap-node/blob/master/src/api/methods/txResponse.js#L38) to please it.

I think it works fine for Transfer transactions, but breaks on most of the other types, because they miss "required" fields. E.g. it fails on Deposit tx because it has empty `from` field etc

Reply:
probably could be an easy fix on the node side https://github.com/leapdao/leap-node/issues/290